zerver/lib/export.py

```diff
@@ -460,10 +454,15 @@ def export_uploads(realm, output_dir):
     export_bucket(realm, settings.S3_AVATAR_BUCKET, os.path.join(output_dir, "avatars"), True)
     export_bucket(realm, settings.S3_AUTH_UPLOADS_BUCKET, os.path.join(output_dir, "uploads"))
 
-def do_export_realm(realm, output_dir, threads=0):
+def do_export_realm(realm, output_dir, threads):
     # type: (Realm, Path, int) -> None
     response = {} # type: TableData
 
+    # We need at least one thread running to export
+    # UserMessage rows.  The management command should
+    # enforce this for us.
+    assert threads >= 1
+
```

Comment:
Yeah.  BTW in theory if we removed the assert you could run the lib code now with zero threads to defer the UserMessage export step.  We'd need to change the management command to allow that as well.